server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java

```diff
@@ -46,9 +48,25 @@ public Invocations getInvocations() {
             .invocation().methods(GET).path("/v2/tasks/").handleWith(this::listTasks)
             .invocation().methods(PUT, POST).path("/v2/tasks/{taskName}").handleWith(this::createScriptTask)
             .invocation().methods(POST).path("/v2/tasks/{taskName}").withAction("exec").handleWith(this::runTask)
+            .invocation().methods(GET).path("/v2/tasks/{taskName}").withAction("script").handleWith(this::getScript)
             .create();
    }
 
+   private CompletionStage<RestResponse> getScript(RestRequest request) {
+      String taskName = request.variables().get("taskName");
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance().withSubject(request.getSubject());
+      ScriptingManager scriptingManager = SecurityActions.getGlobalComponentRegistry(cacheManager).getComponent(ScriptingManager.class);
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+      if (!scriptingManager.getScriptNames().contains(taskName)) {
+         throw Log.REST.noSuchScript(taskName);
+      }
+
+      return CompletableFuture.supplyAsync(() -> {
+         String script = Subject.doAs(request.getSubject(), (PrivilegedAction<String>) () -> scriptingManager.getScript(taskName));
+         return builder.entity(script).build();
```

Comment:
should we add a content-type here or unnecessary?

Reply:
probably I should set it to `text/plain`. In theory we could determine it from the script metadata (which is not exposed by the ScriptingManager API).